### DISCUSSION TITLE
Usage of getPlayerBossBars

### DISCUSSION THREAD

**Discussion by AntonIXO** [Minestom/Minestom#2157] [Help] [answered]
I have following code:
```java
public static void builderBar(Player player) {
        Integer points = player.getInstance().getTag(Tags.points);
        BossBarManager manager = new BossBarManager();
        
        Collection<BossBar> playerBars = manager.getPlayerBossBars(player);
        if (playerBars.isEmpty()) {
            System.out.println("First bar is not empty");
            BossBar firstBar = playerBars.iterator().next();
            firstBar.name(Component.text("World bits: " + points));
            firstBar.progress(points / 100);
        } else {
            BossBar bar = BossBar.bossBar(Component.text("World bits: " + points), 0.6f, BossBar.Color.GREEN, BossBar.Overlay.PROGRESS);
            player.showBossBar(bar);
        }
    }
```
Why it does no working and I could not get if player have bossbar?
I also tried playerBars.stream().findFirst(); and it's also not working

**Comment by BL19** [comment, marked as answer]
It seems like you are creating a new `BossBarManager` every time and I guess that's why it's not working. I haven't tried this, but my guess is that using [MinecraftServer#getBossBarManager()](https://javadoc.minestom.net/net/minestom/server/MinecraftServer.html#getBossBarManager()) instead of `new BossBarManager()` would solve it.

**Reply by AntonIXO** [reply]
You're right. Thanks a lot, spent a few hours on it!
